data_model/src/lib.rs

```diff
@@ -1804,6 +1810,89 @@ pub mod domain {
         }
     }
 
+    /// Represents path in IPFS. Performs some checks to ensure path validity.
+    ///
+    /// Should be constructed with `from_str()` method.
+    #[derive(
+        Debug,
+        Clone,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        Hash,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+    )]
+    pub struct IpfsPath(String);
+
+    impl FromStr for IpfsPath {
+        type Err = ParseError;
+
+        fn from_str(string: &str) -> Result<Self, Self::Err> {
+            let mut subpath = string.split('/');
+            let empty = subpath.next().ok_or(ParseError {
+                reason: "Impossible error",
+            })?;
+
+            if empty.is_empty() {
+                let root_type = subpath.next().ok_or(ParseError {
+                    reason: "Expected root type, but nothing found",
+                })?;
+                let key = subpath.next().ok_or(ParseError {
+                    reason: "Expected at least one content id",
+                })?;
+
+                match root_type {
+                    "ipfs" | "ipld" => Self::check_cid(key)?,
+                    "ipns" => (),
+                    _ => {
+                        return Err(ParseError {
+                            reason: "Unexpected root type",
```

Comment:
```suggestion
                            reason: format!("Unexpected root type. Expected `ipfs`, `ipld` or `ipns`. Got {}", root_type)
```

Reply:
Can't do that, because `reason` should be `static &str`. I don't know why, but I just used that was provided by existing code.
I'll add expected types